src/test/libcephfs/test.cc

```diff
@@ -4148,3 +4148,137 @@ TEST(LibCephFS, InodeGetPut) {
   ASSERT_EQ(0, ceph_unmount(cmount));
   ceph_shutdown(cmount);
 }
+
+static bool write_done = false;
+static bool fsync_done = false;
+static std::mutex mtx;
+static std::condition_variable cond;
+
+void write_fsync_io_callback(struct ceph_ll_io_info *io_info) {
+  std::unique_lock lock(mtx);
+  if (io_info->write) {
+    std::cout << "written=" << io_info->result << std::endl;
+    write_done = true;
+  } else {
+    std::cout << "fsync" << std::endl;
+    fsync_done = true;
+  }
+  cond.notify_one();
+}
+
+static void writer_func(struct ceph_mount_info *cmount, Fh *fh) {
+  int iterations = 2;
+  uint8_t buf[131072];
+  struct ceph_ll_io_info io_info;
+  struct iovec iov;
+
+  io_info.callback = write_fsync_io_callback;
+  io_info.iov = &iov;
+  io_info.iovcnt = 1;
+  io_info.off = 0;
+  io_info.fh = fh;
+  io_info.write = true;
+  io_info.fsync = false;
+
+  while (--iterations > 0) {
```

Comment:
s/`--iterations`/`iterations--` otherwise it would just loop once.

Reply:
Actually that's the intent. I was playing with various iterations and it only took 1 iteration to reproduce this.

Reply:
then the loop can be removed altogether to simplify the test case

Reply:
That's not the point, Dhairya. I needed the number of iterations specified in this change in my local test. It might more in teuthology. Only when this is run through tests is when we will know.

Reply:
oh so the iterations is yet to be finalised. got it.

Reply:
Once I get some test results, I'll check if we need to bump the iterations up a bit.